Assets/HoloToolkit/Input/Scripts/GestureManager.cs

```diff
@@ -256,31 +401,56 @@ void LateUpdate()
                 newFocusedObject = OverrideFocusedObject;
             }
 
+            // Checks to see if our focus has changed.
             bool focusedChanged = FocusedObject != newFocusedObject;
 
-#if UNITY_EDITOR
-            if (Input.GetMouseButtonDown(1) || Input.GetKeyDown(EditorSelectKey))
-            {
-                OnTap();
-                OnRecognitionStarted();
-            }
-
-            if (Input.GetMouseButtonUp(1) || Input.GetKeyUp(EditorSelectKey) || focusedChanged)
-            {
-                OnRecognitionEndeded();
-            }
-#endif
             if (focusedChanged)
             {
                 // If the currently focused object doesn't match the new focused object, cancel the current gesture.
-                // Start looking for new gestures.  This is to prevent applying gestures from one hologram to another.
+                // This is to prevent applying gestures from one hologram to another.
                 gestureRecognizer.CancelGestures();
+
+                // Set our last Focused object.
+                lastFocusedObject = FocusedObject;
+
+                // Set our current Focused Object.
                 FocusedObject = newFocusedObject;
+
+                // Start looking for new gestures.
                 gestureRecognizer.StartCapturingGestures();
             }
+
+            return focusedChanged;
+        }
+
+        private void LateUpdate()
+        {
+            bool focusedChanged = CalcFocusedObject();
+
+#if UNITY_EDITOR || UNITY_STANDALONE
+            // Process Editor/Companion app input.  Tap by pressing both right and left mouse buttons.  Release Tap is on any mouse button up.
+
+            // If we're already pressing a button, our keyboard select key, or if the focus has changed then throw recognition Ended.
+            if (Input.GetMouseButtonUp(0) || Input.GetMouseButtonUp(1) || Input.GetKeyUp(keyboardSelectKey) || focusedChanged)
+            {
+                OnRecognitionEndeded(focusedChanged);
+            }
+
+            // If we're currently pressing both mouse buttons, or our keyboard select key.
+            if ((Input.GetMouseButtonDown(0) && Input.GetMouseButtonDown(1)) || Input.GetKeyDown(keyboardSelectKey))
```

Comment:
I'm not a fan of forcing the two buttons on the mouse to be pressed, it's actually really difficult to do it exactly at the same time. If one button is pressed slightly later than the other, the tap does not register. The right button is what is used in the HoloLens Emulator which uses the right mouse button for clicking. Is there a reason why we need two mouse buttons to click?